.github/workflows/github-actions.yml

```diff
@@ -0,0 +1,37 @@
+name: Tests
+on:
+  push:
+    branches: '**'
+  pull_request:
+    branches: main
+jobs:
+  unit-testing:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        python-version: ['3.10', '3.11', '3.12']
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set Up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Display Python Version
+        run: python -c "import sys; print(sys.version)"
+        shell: bash
+      - name: Upgrade Pip
+        run: python -m pip install --upgrade pip
+        shell: bash
+      - name: Install requirements
+        run: python -m pip install --upgrade -r requirements.txt
+        shell: bash
+      - name: Install other dependencies
+        run: python -m pip install --upgrade pytest black flake8
```

Comment:
No minimum version is specified for these libraries here.

Reply:
Yes

Reply:
I don't even think we need to test so many versions. Instead, is it possible to simply choose a reasonably stable Python version?

Reply:
>  Instead, is it possible to simply choose a reasonably stable Python version?

How about choosing one-before-last [major] version?

Reply:
You might hate this but I wrote an ugly `curl` function that picks the "safe" version of Python using some xpath magic:

```
#!/bin/bash

get_safe_python_version()
{
    SAFE_PYTHON=curl --location https://devguide.python.org/versions | xmllint --html --xpath '//section[@id="supported-versions"]//table/tbody/tr[count(//section[@id="supported-versions"]//table/tbody/tr[td[.="security"]]/preceding-sibling::*)]/td[1]/p/text()' - 2>/dev/null
}

get_safe_python_version
echo $SAFE_PYTHON
```

Basically, it scrapes `https://devguide.python.org/versions`, looks for the latest "security" version (yellow), and takes the Python version that is one less than the latest security version.

Reply:
@seanlaw 
Cool! IIRC, I think I saw, somewhere in STUMPY, that you used `pandas` to read table and it was really clean. So, can this work?

```
import pandas as pd
from packaging.version import Version
df = pd.read_html("https://devguide.python.org/versions/")[0]
out = df[df.Status == 'security'].Branch.tolist()
out = sorted(out, key=lambda x: Version(x), reverse=True)
print(out[1])  # 3.10
```

Reply:
Yes but now this adds a new dependency that requires you to install `pandas` in the Github workflow environment just so you can do this. Whereas the one-liner "just works" out of the box.

Reply:
> nvm. Because now I think I need a way to return that value and assign it to a variable in shell.

Exactly! This (ugly) one liner will get you there immediately

Reply:
Yeah...And the "trade-off" you mentioned makes sense here as we most likely do not need to revisit it in the future.

Reply:
Thanks for sharing the `curl`! Going to revise Github Actions.